scratch.Rmd

```diff
@@ -0,0 +1,556 @@
+}
+```
+
+# Downloading relevant outside datasets
+
+```{r}
+usgs_lincoln <- dataRetrieval::readNWISuv("06752260", parameterCd = "00065", startDate = "2019-09-01", endDate = "2022-12-01") %>%
+  mutate(DT=ymd_hms(dateTime))%>%
+  rename(Depth_ft = X_00065_00000) %>%
+  #mutate(date = as_date(dateTime)) %>%
+  #group_by(date) %>%
+  #summarize(X_00065_00000 = mean(X_00065_00000)) %>%
+  #mutate(Depth_m = X_00065_00000 * 0.3048) %>%
+  select(DT, 
+         Depth_ft) %>%
+  mutate(site="usgs_lincoln")
+```
+
+
+# Legacy
+
+So far in 2022, all data has been sent to HydroVu (outside of genomic sampling date 2022-09-19). This location encountered several issues this field season: sensor pulled 2022-05-24 due to fears that its infrastructure would wash away. Sensor was re-deployed 2022-06-01, but was then pulled out due to tech issues 2022-07-08 and turbidity problems. Sometime between then and 2022-07-12, turbidity sensor totally broke. 2022-07-16, the turbidity sensor was replaced with the one that was previously being used at Archery. Between 2022-07-18 and 2022-07-21, the back-up sensor stopped working. After this we we swapped the sonde with sensor from Rist 2022-08-03. On 2022-08-04, the sensor was pulled because flows were too low (sensor was likely not suspended in the water 8/3 - 8/4). Redeployed on 2022-08-25, though issues with turbidity and ORP occurred from then until
+
+```{r}
+# No VuLink connection during final genomic survey (water was too low for full deployment)
+raw_troll_legacy <- map_dfr(grep(list.files("data/sensor_data/2022/legacy/manual/", full.names = T), pattern = "trolled", invert = F, value = T), troll_reader)
+names(raw_troll_legacy) <- make.names(names(raw_troll_legacy), unique = T)
+
+rawless_troll_legacy <- raw_troll_legacy %>% 
+  select(DT = contains('Date.Time'),
+                      Water_Temp_C = as.numeric(contains('Temperature...C')),
+                      pH = contains('pH'),
+                      ORP_mV = contains('ORP'),
+                      #Actual_Conductivity_µS_cm = contains('Actual.Conductivity..µS.cm.'),
+                      Specific_Conductivity_µS_cm = contains('Specific.Conductivity..µS.cm.'),
+                      #Salinity_PSU = contains('Salinity..psu.'),
+                      DO_ppm = contains('RDO.concentration'),
+                      Turbidity_NTU = contains('Turbidity'),
+                      #TSS_mg_L = contains('Total.Suspended.Solids..mg.L.'),
+                      #Chla_RFU = contains('Chl.a.Fluorescence..RFU.'),
+                      #Chla_µg_L = contains('Chl.a.Concentration..µg.L.'),
+                      #Pressure_PSI = contains('Pressure..psi.'),
+                      Depth_ft = contains('Depth..ft')) %>%
+                      #Elevation_m = contains('Level..Elevation..m.'),
+                      #Depth_to_Water_ft = contains('Level..Depth.to.Water..ft.')) %>%
+  mutate(DT = ymd_hms(DT)) %>%
+  mutate(DT = DT - lubridate::hours(1)) %>% # reported in MDT >:(
```

Comment:
Note that you can store the `ymd_hms` with a tz argument, then use `with_tz()` to convert datetime stamps to another timezone. This is especially helpful when you're dealing with data that have been converted on DST beyond DST start/end or when you do happen to have data outside of DST that need to be managed as UTC datetimes so you don't eliminate data with overlapping time stamps in the fall change. This is the most easily-read list of tz's: https://en.wikipedia.org/wiki/List_of_tz_database_time_zones; where you'd use 'TZ data base name' as your tz argument in `ymd_hms()`